>>> mul = __import__('1-mul').mul


Testing Integers:

>>> mul(2, 2)
4
>>> mul(-3, -3)
9
>>> mul(-1, 3)
-3
>>> mul(1, -3)
-3
>>> mul(110 - 10, 3)
300
>>> mul(1, 110 - 10)
100


Testing Floats:

>>> mul(2.5, 4)
10.0
>>> mul(4, -2.5)
-10.0
>>> mul(-2.5, -4.0)
10.0
>>> mul(float("inf"), 2)
inf
>>> mul("hi", 2.1)
Traceback (most recent call last):
...
TypeError: can't multiply sequence by non-int of type 'float'


Testing Lists, Tuples, Dictionaries, and Sets:

>>> mul([1, 2], 2)
[1, 2, 1, 2]
>>> mul(2, [1, 2])
[1, 2, 1, 2]
>>> mul((1, 2), (1, 2))
Traceback (most recent call last):
â€¦
TypeError: can't multiply sequence by non-int of type 'tuple'
>>> mul([1, 2], [2, 1])
Traceback (most recent call last):
...
TypeError: can't multiply sequence by non-int of type 'list'
>>> mul(10, {0: "hi"})
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'int' and 'dict'
>>> mul(('hi', 3), 2)
('hi', 3, 'hi', 3)
>>> mul({'f', 'u', 'c', 'k'}, 1)
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'set' and 'int'


Testing Booleans:

>>> mul(True, 3)
3
>>> mul(False, 3)
0
>>> mul(3, True)
3
>>> mul(3, False)
0
>>> mul(True, False)
0

Testing Characters and Strings:

>>> mul(3, 'a')
'aaa'
>>> mul('b', 3)
'bbb'
>>> mul("hi", 3)
'hihihi'
>>> mul(1, "hello")
'hello'
>>> mul("holy", "shit")
Traceback (most recent call last):
...
TypeError: can't multiply sequence by non-int of type 'str'


Testing None

>>> mul(1, None)
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'
>>> mul(None, 2)
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'

>>> mul(None, None)
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'NoneType' and 'NoneType'
>>> mul(7, lambda x: x)
Traceback (most recent call last):
...
TypeError: unsupported operand type(s) for *: 'int' and 'function'


Testing Invalid Arguments:

>>> mul(1)
Traceback (most recent call last):
...
TypeError: mul() missing 1 required positional argument: 'b'
>>> mul(1, 1, 1)
Traceback (most recent call last):
...
TypeError: mul() takes 2 positional arguments but 3 were given
>>> mul()
Traceback (most recent call last):
...
TypeError: mul() missing 2 required positional arguments: 'a' and 'b'
